src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -293,7 +272,33 @@ try_append_module(sentry_value_t modules, const sentry_module_t *module)
         return;
     }
 
-    sentry_value_append(modules, sentry__procmaps_module_to_value(module));
+    // At least on the android API-16, x86 simulator, the linker apparently
+    // does not load the complete file into memory. Or at least, the section
+    // headers which are located at the end of the file are not loaded, and
+    // we would be poking into invalid memory. To be safe, we load the file
+    // from disk, so we have the on-disk layout, and are independent of how the
+    // runtime linker would load or re-order any sections. The exception here is
+    // the linux-gate, which is not an actual file on disk, so we actually poke
+    // at its memory.
+    sentry_value_t mod = sentry_value_new_null();
+
+    if (sentry__slice_eq(module->file, LINUX_GATE)) {
+        mod = sentry__procmaps_module_to_value(module, module->start);
+    } else {
```

Comment:
This means we're now copying every file into memory now on every platform. This is potentially a *lot* and might seriously halt the application for a while. Can we potentially detect that the module is not loaded properly and only do this conditionally? Alternatively, could we memory-map here to reduce the impact of this a bit?

Reply:
I would prefer mmap, since the OS should be smart enough to not create a separate copy for the already mapped libraries. I will try that tomorrow.

Reply:
Sounds good. Even though mmap introduces some complexity, I think it's well worth it. Thanks!